api/src/server.rs

```diff
@@ -817,6 +832,142 @@ where
         let stop = stop_builder.with_max_event().build_fencepost();
         Ok((start, stop))
     }
+
+    async fn get_stream_state(
+        &self,
+        stream_id: String,
+    ) -> Result<ceramic_api_server::StreamsStreamIdGetResponse, ErrorResponse> {
+        let stream_id = StreamId::from_str(&stream_id)
+            .map_err(|err| ErrorResponse::new(format!("invalid stream_id: {err}")))?;
+        let state_batch = self
+            .pipeline
+            .table("doc_state")
+            .await
+            .map_err(|err| ErrorResponse::new(format!("doc_state table not found: {err}")))?
+            .select(vec![
+                col("stream_cid"),
+                col("event_cid"),
+                col("dimensions"),
+                col("controller"),
+                col("state"),
+                col("index"),
+            ])
+            .map_err(|err| ErrorResponse::new(format!("failed to select: {err}")))?
+            .filter(col("stream_cid").eq(lit(stream_id.cid.to_bytes())))
+            .map_err(|err| ErrorResponse::new(format!("failed to filter: {err}")))?
+            .aggregate(
+                vec![col("stream_cid"), col("controller")],
+                vec![
+                    last_value(vec![col("state")])
```

Comment:
what is the "state" column?  Shouldn't this be "data" or "content"?

Reply:
State is the name of the column that contains both the metadata and the data. We do no use the word content anywhere.

Reply:
I wonder if we should rename what we call "state" to "data", and what we call "data" to "content"?

Reply:
This is effectively what I propose above. Thanks for the suggestion.